runtime_scan/pkg/provider/azure/client.go

```diff
@@ -222,19 +223,48 @@ func processVirtualMachineListIntoAssetTypes(vmList armcompute.VirtualMachineLis
 	return ret, nil
 }
 
-func getVMInfoFromVirtualMachine(vm *armcompute.VirtualMachine) (models.AssetType, error) {
+type rootVolumeInfo struct {
+	SizeGB    int32
+	Encrypted bool
+}
+
+func (c *Client) getRootVolumeInfo(ctx context.Context, vm *armcompute.VirtualMachine) *rootVolumeInfo {
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+	ret := &rootVolumeInfo{
+		SizeGB:    *vm.Properties.StorageProfile.OSDisk.DiskSizeGB,
+		Encrypted: false,
+	}
+	osDiskID, err := arm.ParseResourceID(*vm.Properties.StorageProfile.OSDisk.ManagedDisk.ID)
+	if err != nil {
+		logger.Warnf("Failed to parse disk ID. DiskID=%v: %v", *vm.Properties.StorageProfile.OSDisk.ManagedDisk.ID, err)
+		return ret
+	}
+	osDisk, err := c.disksClient.Get(context.TODO(), osDiskID.ResourceGroupName, osDiskID.Name, nil)
+	if err != nil {
+		logger.Warnf("Failed to get OS disk. DiskID=%v: %v", *vm.Properties.StorageProfile.OSDisk.ManagedDisk.ID, err)
+		return ret
```

Comment:
I like the idea of not failing to discover the asset if we can't get its root volume info, however I'm not sure I like that we're setting Encrypted to false in this case. Perhaps encrypted should be an enum with three states, yes, no, and unknown?

Reply:
I like the idea but I think that it will raise questions what unknown means and what needs to be done for it to be known.
I think that we need to better define it (e.g. store and show the error why it's unknown). Maybe better to do in a separate task?

Reply:
I'm just concern that people might try to run a scan by filtering on whether the disk is encrypted or not, if we set it to "false" if we failed to determine it then they might get an error like "Failed to snapshot disk because its encrypted", but then our API says that the root disk is encrypted.

I'd rather be truthful and say and we were unable to determine it, perhaps unknown is the wrong word, maybe "undetermined"?

Reply:
Done